api_client/src/lib.rs

```diff
@@ -127,7 +127,10 @@ fn parse_http_response(socket: &mut dyn Read) -> Result<Option<String>, Error> {
             }
         }
     }
-    let body_string = content_length.and(Some(String::from(&res[body_offset.unwrap()..])));
+    let body_string = match body_offset {
+        Some(body_offset) => content_length.and(Some(String::from(&res[body_offset..]))),
+        None => None,
+    };
```

Comment:
This can be simplified with the following:
```rust
let body_string = content_length.and(body_offset.map(|o| String::from(&res[o..])));
```

Reply:
Thank you for the suggestion. I will apply it.